src/blur.rs

```diff
@@ -0,0 +1,52 @@
+use image::{ImageFormat, RgbaImage};
+use rocket::response::Content;
+use image::io::Reader;
+use std::io::Cursor;
+use rocket::http::ContentType;
+use gif::{Encoder, Frame, Repeat, DisposalMethod};
+use crate::payload::IntoVec;
+
+pub fn blur_base_on_type(data: &Vec<u8>, image_reader: Reader<Cursor<&Vec<u8>>>, format: ImageFormat, sigma: f32) -> Content<Vec<u8>> {
+    return match format {
+        ImageFormat::Gif => {
+            let mut decoder = gif::DecodeOptions::new();
+            decoder.set_color_output(gif::ColorOutput::RGBA);
+            let mut decoder = decoder.read_info(Cursor::new(data)).unwrap();
+
+            let width = decoder.width();
+            let height = decoder.height();
+
+            let mut frames: Vec<Vec<u8>> = vec!();
+
+            while let Some(frame) = decoder.read_next_frame().unwrap() {
+                let frame: &Frame = frame;
+
+                let mut vec = vec!();
+                for s in frame.buffer.iter() {
+                    vec.push(*s);
+                }
+                frames.push(vec)
+            }
+            let buffer_size = width as usize * height as usize * frames.len() * 2;
+            let mut vec1 = vec![0; buffer_size];
+            {
+                let mut encoder = Encoder::new(vec1.as_mut_slice(), width, height, &[]).unwrap();
+                encoder.set_repeat(Repeat::Infinite).expect("oh no, couldn't set gif to repeat");
```

Comment:
Shouldn't you aim to reproduce as much of the settings as the original gif? Try to query the `repeat` of the original gif and replicate it here.

In the future, some kind of 'desired state' could be an argument to function to (optionally) overwrite this behaviour.